enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,455 @@
+        "message": "violation - configmaps [app-data] not found in namespace default",
+        "timestamp": "2023-07-19T18:25:43.511Z",
+        "metadata": {}
+      }
+    },
+    {
+      "cluster": {
+        "name": "cluster2"
+      },
+      "parent_policy": {
+        "name": "policies.etcd-encryption"
+      },
+      "policy": {
+        "apiGroup": "policy.open-cluster-management.io",
+        "kind": "ConfigurationPolicy",
+        "spec": {},
+        "specHash": {}
+      },
+      "event": {
+        "compliance": "NonCompliant",
+        "message": "notification - configmaps [app-data] found as specified in namespace default",
+        "timestamp": "2023-07-19T18:25:41.523Z",
+        "metadata": {}
+      }
+    }
+  ],
+  "metadata": {
+    "page": 1,
+    "per_page": 20,
+    "total": 123
+  }
+}
+```
+
+##### Generating a Report
+
+Another HTTP endpoint would be added at `/api/v1/reports/compliance-events` that would accept a `GET` request with the
+same query parameters as the `/api/v1/compliance-events` HTTP endpoint (excluding pagination), but would generate a CSV
+file in the same format as the `/api/v1/compliance-events` HTTP endpoint, but flattened. For example,
+`{"parent_policy": {"name": "policies.etcd-encryption"}}` would become `parent_policy_name` as the column name.
+
+##### Managed Cluster Phase 1
+
+The minimal approach is to have the Status Sync controller also record the compliance event in the database on the hub
+when it's recording the compliance event on the parent replicated policy. This phase 1 approach would not be resilient
+to lost connectivity to the hub since there would not be a compliance event queue on the managed cluster. This would be
+addressed in phase 2.
+
+The current compliance events sent by the controllers is missing some metadata. To resolve this, the events will set the
+`related.resourceVersion` field to the `resourceVersion` of the policy that was evaluated (e.g. a
+`ConfigurationPolicy`). Additionally, events will have a new annotation of
+`policy.open-cluster-management.io/policy-evaluated-generation` set to the `generation` of the policy that was
+evaluated. With this additional information, the Status Sync controller can look up the policy by the `resourceVersion`,
+and if by the time the lookup occurs the policy with that resource version is no longer in etcd, the controller will
+lookup the current version of the policy and see if the generation from the annotation matches. Then the spec from that
+lookup will be recorded in the compliance event. If the spec couldn't be determined, a value of `unknown` will be
+recorded.
+
+Note, to reduce bandwidth consumption, the Status Sync controller should cache a list of spec hashes and their
+associated `generation` in memory as a compliance event is recorded. Then, this cache can be used to determine if the
+full spec needs to be retrieved by the Status Sync controller and provided to the `POST` request on the
```

Comment:
I guess I'm wondering if there's a difference between the hub managed cluster replicated spec and the spec that the managed cluster is sending here in the POST.

Reply:
This will be covered in the conversation in https://github.com/open-cluster-management-io/enhancements/pull/98#pullrequestreview-1548522146.